### ISSUE TITLE
bug(developer): inconsistency between 13.0 and 14.0 for BCP 47 tag canonicalisation

### ISSUE THREAD

**Issue by MakaraSok** [keymanapp/keyman#4509]
**Describe the bug**

The language tag has been set to 'mhi-Latn', but the Developer does not compile and ask for only 'mhi'.

Error message in Developer:
```
sil_madi.kps: Compiling package sil_madi.kps...
sil_madi.kps: Warning: Keyboard with ID 'sil_madi' has a BCP 47 warning: 'mhi-Latn' is a valid tag but is not canonical: it should be 'mhi'.
sil_madi.kps: Failure: 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\source\sil_madi.kps' was not compiled successfully.
```

The keyboard get compiled successfully though in bash when the tag is 'mhi-Latn'.
```
makara@AORAL MINGW64 /c/keyman/keyboards/keyboards (sil_madi)
$ ./build.sh release/sil/sil_madi
--- Only building release release/sil/sil_madi ---
Building sil_madi
sil_madi.keyboard_info: Success: File sil_madi.keyboard_info validated successfully.
Building keyboard release/sil/sil_madi

sil_madi.kmn: Compiling 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\source\sil_madi.kmn' for Windows, macOS, Linux, Desktop devices...
sil_madi.kmn: Success: 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\source\sil_madi.kmn' was compiled successfully for Windows, macOS, Linux, Desktop devices to 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\build\sil_madi.kmx'.
sil_madi.kmn: Compiling 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\source\sil_madi.kmn' for Web, iPhone, iPad, Android phone, Android tablet, Mobile devices, Tablet devices...
sil_madi.kmn: Success: 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\source\sil_madi.kmn' was compiled successfully for Web, iPhone, iPad, Android phone, Android tablet, Mobile devices, Tablet devices to 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\build\sil_madi.js'.
sil_madi.kps: Compiling package sil_madi.kps...
sil_madi.kps: Success: 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\source\sil_madi.kps' compiled successfully to 'C:\keyman\keyboards\keyboards\release\sil\sil_madi\build\sil_madi.kmp'.
Merging sil_madi.keyboard_info for sil_madi
```

However, if the language tag is set to 'mhi' to satisfy the Developer, the keyboard won't compile in bash.

Developer:
![image](https://user-images.githubusercontent.com/28331388/108831781-2f8ede80-75fd-11eb-9e1d-3209f8c69145.png)

Bash:
![image](https://user-images.githubusercontent.com/28331388/108831856-47666280-75fd-11eb-974e-92de0d5c21cc.png)

**To Reproduce**

1. Open this keyboard '[sil_madi.zip](https://github.com/keymanapp/keyman/files/6028314/sil_madi.zip)'
2. Click on the keyboard package
3. Compile the package see if successfully compile
4. Now open git bash and compile it from the baseline
5. See error 

**Expected behavior**

When the language tag is set to "mhi-Latin", the keyboard should be compiled successfully in both the Developer and the baseline.

---

**Keyman for Windows:**
 - OS: Windows 10 Pro 20H2 x64
 - Keyman version: 14.0.239-beta
 - Target Application: Developer

**Keyman Developer:**
 - OS: Windows Pro 10 2004 x64
 - Keyman Developer version: 14.0.239-beta

---

**Keyboard**
 - Keyboard name: sil_madi
 - Keyboard version: 1.0 (in development)
 - Language name: Ma'di

---

**Additional Contexts**
Neither "mhi-Latn" nor "mhi" tag would satisfy both compilers consistently.

**Comment by mcdurdin** [comment]
This relates to how the compiler treats BCP47 codes for 14.0. Because we have a new source of truth for canonicality of these codes (langtags.json), the canonicalisation has changed fairly significantly.

The command line compiler in the keyboards repository is not yet using 14.0 code -- that's why we see the discrepancy.

I am having a discussion next week on this problem.

**Comment by jahorton** [comment]
> This relates to how the compiler treats BCP47 codes for 14.0. Because we have a new source of truth for canonicality of these codes (langtags.json), the canonicalisation has changed fairly significantly.

So... is our, uh, "canonicality source" canonical this time?

**Comment by mcdurdin** [comment]
> So... is our, uh, "canonicality source" canonical this time?

For some level of canonicality, yes. There is a greater degree of canonicalisation now as there is a whole lot more data. But the canonicalisation will change over time. That's one of my concerns.

**Comment by MakaraSok** [comment]
The keyboard's PR is now up and, as expected, not compile successfully. https://github.com/keymanapp/keyboards/pull/1472

![image](https://user-images.githubusercontent.com/28331388/109261761-ecb64c00-7832-11eb-8a47-40db572e65f1.png)

**Comment by mcdurdin** [comment]
This will be resolved with #4563 as the warning is downgraded to a hint for version 14.0.
